packages/ui-react/src/components/Image/Image.tsx

```diff
@@ -16,14 +16,56 @@ const setWidth = (url: string, width: number) => {
   return createURL(url, { width });
 };
 
+const cache = new Map();
+
+const resolveImageURL = async (imgUrl: string, width: number) => {
```

Comment:
Is it something we need because of the thumbnail fallback returning 1920 image?

Reply:
Yes, that's correct. But also because we found that the redirect causes the browser caching mechanism to fail. Not sure if this is caused by invalid/missing cache headers, but this fix resolves the redirect URL once and caches it for the next time.

Reply:
We have open PRs to fix both now (adding Cache-Control header + fallback image size fix). Hope it will be delivered soon.

Reply:
That would be awesome! Do you have an ETA? 

I'm asking because currently the homepage images are loading really slow in different regions (Thailand for example). This fix drastically improves that, so we could decide to merge this and remove it when the CDN PRs are landed?

Reply:
Yeah, we can also merge these changes before. I hope this / next week.